client-side/src/App.jsx

```diff
@@ -13,9 +15,12 @@ function App() {
     <>
       <SnackbarProvider maxSnack={3}>
       <Provider store={store}>
+      <PopupProvider>
```

Comment:
why does the popupProvider wraps the router & footer? Shouldn't it wrap only the popup itself?